src/blockchaintree.rs

```diff
@@ -852,14 +858,17 @@ impl BlockChainTree {
             .change_context(BlockChainTreeError::BlockChainTree(
                 BCTreeErrorKind::CreateDerivChain,
             ))?;
-
+        
         chain
             .dump_config(&root_path)
             .change_context(BlockChainTreeError::BlockChainTree(
                 BCTreeErrorKind::CreateDerivChain,
             ))?;
 
-        Ok(Box::new(chain))
+        self.deratives.insert(*addr, Arc::new(RwLock::new(chain)));
+        
+
+        Ok(self.deratives.get(addr).unwrap())
```

Comment:
Better to use [entry](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry)